src/Microsoft.AspNet.Http.Abstractions/Authentication/AuthenticationManager.cs

```diff
@@ -11,41 +13,66 @@ public abstract class AuthenticationManager
     {
         public abstract IEnumerable<AuthenticationDescription> GetAuthenticationSchemes();
 
-        public abstract AuthenticationResult Authenticate(string authenticationScheme);
+        public abstract Task AuthenticateAsync([NotNull] AuthenticateContext context);
 
-        public abstract Task<AuthenticationResult> AuthenticateAsync(string authenticationScheme);
+        public async Task<ClaimsPrincipal> AuthenticateAsync([NotNull] string authenticationScheme)
+        {
+            var context = new AuthenticateContext(authenticationScheme);
+            await AuthenticateAsync(context);
+            return context.Principal;
+        }
 
-        public virtual void Challenge()
+        public virtual Task ChallengeAsync()
         {
-            Challenge(authenticationScheme: null, properties: null);
+            return ChallengeAsync(authenticationScheme: null, properties: null);
         }
 
-        public virtual void Challenge(AuthenticationProperties properties)
+        public virtual Task ChallengeAsync(AuthenticationProperties properties)
         {
-            Challenge(authenticationScheme: null, properties: properties);
+            return ChallengeAsync(authenticationScheme: null, properties: properties);
         }
 
-        public virtual void Challenge(string authenticationScheme)
+        public virtual Task ChallengeAsync([NotNull] string authenticationScheme)
         {
-            Challenge(authenticationScheme: authenticationScheme, properties: null);
+            return ChallengeAsync(authenticationScheme: authenticationScheme, properties: null);
         }
 
-        public abstract void Challenge(string authenticationScheme, AuthenticationProperties properties);
+        // Leave it up to authentication handler to do the right thing for the challenge
+        public Task ChallengeAsync([NotNull] string authenticationScheme, AuthenticationProperties properties)
```

Comment:
Updated PR to use [NotNull] everywhere, Challenge sugar methods now pass string.Empty since we were already doing string.IsNullOrWhitespace checks, this shouldn't affect anything.  Also nuked the overloads of SignIn/SignOut that didn't require a scheme, this was on the todo list from earlier API reviews with Lou